functions.py

```diff
@@ -1,49 +1,141 @@
 
 #how much is not taxed
-def SellWithoutTax():
-    arrayline = ReadLines('new_register.txt')
+def SellWithoutTax(arrdata, price):
     sum = 0.0
-    i = 0
-    arr = arrayline[i].split(',', 2)
-    while (time.time() - float(arr[1])) >= 31536000.0:
-        sum += float(arr[0])
-        i += 1
-        arr = arrayline[i].split(',', 2)
-    print("can sell without tax:", sum)
-    return sum, i
-
-#benefit from not taxed
-def Benefit(sum, firsttax,price):
-    arrayline = ReadLines('new_register.txt')
-    totalprice = 0.0
-    for i in range(0, firsttax):
-        arr = arrayline[i].split(',', 2)
-        totalprice += float(arr[0]) * float(arr[2])
-    print("had payed:", totalprice)
-    print("will get:", sum * price)    
-    print("benefit:", sum*price - totalprice)
-    return (sum*price - totalprice)
-
-
-#add new line
-def AddLine():
+    for i in range(0, len(arrdata)):
+        if (time.time() - float(arrdata[i].data)) >= SEC_IN_YEAR or float(arrdata[i].price) > price:
+            sum += float(arrdata[i].quantity)
+    return sum
+
+#counting benefit 
+def Benefit(sum, price, arrdata):
+    cantsell = sum
+    hadpaid = 0.0
+    taxable = 0.0
+    for i in range(0, len(arrdata)):
+        if float(sum) == 0.0:
+            break
+        else:
+            if float(arrdata[i].quantity) >= cantsell:
+                hadpaid += cantsell*float(arrdata[i].price)
+                if (time.time() - float(arrdata[i].data)) <= SEC_IN_YEAR and float(arrdata[i].price) <= price:
+                    taxable += cantsell*price
+                cantsell = 0.0
+                break
+            else:
+                hadpaid += arrdata[i].purchase_amount()
+                if (time.time() - float(arrdata[i].data)) <= SEC_IN_YEAR and float(arrdata[i].price) <= price:
+                    taxable += float(arrdata[i].quantity) * price
+                cantsell -= float(arrdata[i].quantity)
+    if cantsell != 0.0:
+        print("insufficient funds, missing", cantsell, "BTC")
+        return False
+    else:
+        print("had payed:", hadpaid)
+        print("will get:", sum*price)    
+        print("benefit:", sum*price - hadpaid)
+        if taxable != 0.0:
+            print("with tax:", taxable)
+        return (sum*price - hadpaid), taxable
+    
+#add new line about the new purchase
+def PurchaseNewLine(nameoffile):
     text = []
     print("number of coins:")
     text.append(input())
     text.append(str(time.time()))
     print("price:")
     text.append(input())
-    print(text)
-    f = open('register_copi.txt','a')
-    text = ", ".join(text)
-    text += "\n"
-    print(type(text))
-    f.write(text)
-    f.close()
-    print("ok")
\ No newline at end of file
+    with open(nameoffile,'a') as f:
+        text = ", ".join(text)
+        print(text)
+        text = "\n" + text
+        print(type(text))
+        f.write(text)
+        f.close()
+    print("ok")
+
+def SellDelLines(arrdata, price, nameoffile):

```

Comment:
arrdata - state 
and docs pls - what are the params , what this function outputs etc